### ISSUE TITLE
Filter out all values <= 0°C

### ISSUE THREAD

**Issue by dbrgn** [gfroerli/api#136]
We already filter this at the relay level, but some historic data ist still part of the database.

This could easily be filtered out in all SQL queries we're calling (`WHERE temperature > 0`):

- Latest measurements
- Historic aggregate stats
- Maybe others?

**Comment by dbrgn** [comment]
For testing: Check out the Tribschen Sensor. It shows historic minimal values at -0.1°C.

**Comment by rnestler** [comment]
Like removing it from the DB or only filter it in the queries?

**Comment by dbrgn** [comment]
Hm, I think filtering is probably better... Or do you think removing the values would be better (e.g. with an admin command)?

Something else I thought of: In case we ever deploy an ice-proof sensor in a lake that occasionally freezes (e.g. Sihlsee), we'd need to adjust the logic 🙂

**Comment by rnestler** [comment]
> Something else I thought of: In case we ever deploy an ice-proof sensor in a lake that occasionally freezes (e.g. Sihlsee), we'd need to adjust the logic 🙂

Well if the ice get's 80cm thick, then we need to adjust the logic :laughing: otherwise the water below the ice will still have above zero temperature.

**Comment by dbrgn** [comment]
Yeah, that's the thing, the ice *does* sometimes get that thick near the shallow shore up there 😄

**Comment by dbrgn** [comment]
@rnestler filtering out values <=0°C would probably be pretty trivial right? Would be nice to have less weird-looking stats without much effort 🙂

**Comment by dbrgn** [comment]
After some more thought: Ideally we'd simply not store newly submitted measurements if they're <0. For existing values, we could write a one-time migration to delete all values below 0. In that case, we don't need to include any filtering for all the queries we have.

**Comment by dbrgn** [comment]
You're right, 😅 In that case, could we do a one-time migration to clean historic values from the DB?

**Comment by rnestler** [comment]
> could we do a one-time migration to clean historic values from the DB?

I'll just execute a one-liner in the rails shell 👍 
```
irb(main):007> Measurement.where(temperature: ...0).first
=>
#<Measurement:0x00007f78c5ce9600
 id: 187029,
 temperature: -0.1,
 custom_attributes: nil,
 sensor_id: 9,
 created_at: Fri, 16 Jun 2023 15:33:15.413142000 UTC +00:00,
 updated_at: Fri, 16 Jun 2023 15:33:15.413142000 UTC +00:00>

irb(main):008> Measurement.where(temperature: ...0).pluck(:sensor_id).uniq
=> [9]

irb(main):010> Measurement.where(temperature: ...0).count
=> 1908

irb(main):011> Measurement.where(temperature: ...0).pluck(:temperature).uniq
=> [-0.1]
```

There seems to be only one sensor affected, where 1908 values are `-0.1`. I'll destroy them.

**Comment by dbrgn** [comment]
Perfect, now the min value is 3.5 degrees! Thanks.
